automated install/basic-install.sh

```diff
@@ -828,8 +828,11 @@ It is also possible to use a DHCP reservation, but if you are going to do that,
 
 # Configure networking via dhcpcd
 setDHCPCD() {
-    # Check if the IP is already in the file
-    if grep -q "${IPV4_ADDRESS}" /etc/dhcpcd.conf; then
+    # Regex for matching a non-commented static ip address setting
+    local regex="[ ]*static ip_address[ ]*=[ ]*${IPV4_ADDRESS}"
```

Comment:
I think the regex needs to be tweaked a bit.

`[ ]*` would only match a space character. The manpage for [`dhcpcd.conf`](https://manpages.ubuntu.com/manpages/focal/man5/dhcpcd.conf.5.html) says that:
>The first word on the line is the option and the rest of the line is the value.
     **Leading and trailing whitespace for the option and value are trimmed.**

Reply:
And I don't think spaces are allowed around the `=` but I don't have a setup to check that right now.

Reply:
I also looked at the same document and couldn't find any other information anywhere about allowed spaces. I added the multiple `[ ]*` in case it may work, just to be on the safe side, but I myself only tested it without the spaces.

I can test it with spaces in a few days or remove the  `[ ]*`  from the regex if you want. In this case we can be sure that a valid static IP setting is configured.

Reply:
`[ ]*` only matches a literal space character. I understand 'whitespace' to include tabs. 

The first check _should_ start with a newline anchor and then zero or more whitespace characters. I don't know if the checks around `=` should include tabs or not. I don't _think_ it's a problem to include a space check around `=`, I'm just mentioning it in case others have insight about the case.

Reply:
And I guess it depends on how **option** and **value** is interpreted. I think `static` is the **option** and `ip_address=ip.ad.dd.r` is the **value**. 

I guess `static ip_address` could be the **option** and `ip.ad.dd.r` is the **value** and `=` separates them?

Reply:
Personally I think that `static ip_address` being the option and `ip.ad.dd.r` being the value makes more sense, since `=` is often used to seperate key value pairs. Also you are definitely right with the whitespaces not only being spaces and new line anchoring (I had it in the beginning but somehow it got lost during testing).

Based on the above assumtion I would rewrite the regex like this: 
`regex="^[ \t\r\n\f]*static ip_address[ \t\r\n\f]*=[ \t\r\n\f]*${ip}[ \t\r\n\f]*$"`

Reply:
Correction:

Since grep -xq only looks at a text line by line it's enought to check for white spaces and tabs. 

`regex="^[ \t]*static ip_address[ \t]*=[ \t]*{IPV4_ADDRESS}[ \t]*$"`

Reply:
Actually, checking for ony tabs and spaces should be enough, since a new option/value pair must start on a new line:
`regex="^[ \t]*static ip_address[ \t]*=[ \t]*${IPV4_ADDRESS}[ \t]*$"`

Reply:
I'm not sure if this is relevant, but looking at 3 different manpages makes it more confusing:
- https://manpages.ubuntu.com/manpages/trusty/man5/dhcpcd.conf.5.html (already linked above)
- https://manpages.debian.org/testing/dhcpcd5/dhcpcd.conf.5.en.html;
- https://linux.die.net/man/5/dhcpd-options.

The last one shows spaces around the equal sign. The other 2 don't.

Also, there are a lot of places talking about using spaces (or commas) as separators.

Reply:
The only place that really matters is Raspberry Pi OS/Raspbian (or what ever it's currently called.) That is the only case that triggers the `dhcpcd.conf` detection. We stopped doing any kind of IP addressing with every other OS and it was my request that we keep holding hands with Raspbian. 

I don't think including spaces/tabs around the `=` is a problem, it allows for more cases than may be needed but it doesn't exclude any valid configurations. 

I would like to keep the `^` anchor for the newline, if only for making the code self-documenting.

Reply:
I agree with including spaces/tabs around the  `=`.

Keeping the `^ ... $` for self documentation is a good point. It doesn't change anything.

Regarding changing `[ \t]*` to `\s*`: I intentionally avoided it since in some sources I read that `\s` apparently shouldn't be supported with the flavor of regex used by grep. (However the regex still worked with `\s` on my local Linux machine.)

http://www.greenend.org.uk/rjk/tech/regexp.html 
https://stackoverflow.com/questions/1074941/what-flavour-of-regular-expression-is-grep

Reply:
>I agree with including spaces/tabs around the =.

Sounds good.

> Keeping the ^ ... $ for self documentation is a good point. It doesn't change anything.

Yeah, consider it stylistic.

> Regarding changing [ \t]* to \s*: I intentionally avoided it

Okay, that makes sense to me. `\s` would add newlines but your earlier comments show that newlines do not need to be considered.

Reply:
I updated the regex/commit accordingly.